packages/strapi/lib/core/bootstrap.js

```diff
@@ -295,23 +295,25 @@ module.exports = function(strapi) {
     hostname = 'localhost';
   }
 
-  // proxy settings
-  const proxy = _.get(strapi.config.currentEnvironment, 'server.proxy', {});
-  strapi.config.proxy = proxy;
-
-  // check if proxy is enabled and construct url
-  strapi.config.url = proxy.enabled
-    ? getURLFromSegments({
-        hostname: proxy.host,
-        port: proxy.port,
-        ssl: proxy.ssl,
-      })
-    : getURLFromSegments({
-        hostname,
-        port: strapi.config.port,
-      });
+  let serverUrl = _.trim(
```

Comment:
This is kinda hard to read can split the logic and add some error messages when the urls are invalid. something like this

```js
if (_.has(conf, 'server.url') {
  try {
  return  _.trim(new URL(conf.server.uril).toString())
 } catch (error) {
  throw new Error('Invalid server.url config. Make sure ....')
 }
}
---



```